packages/core/src/services/documents/scan.ts

```diff
@@ -12,33 +12,13 @@ import {
 } from '../../lib'
 import { DocumentVersionsRepository } from '../../repositories'
 
-/**
- * This is an internal method. It should always receives
- * workspaceId from a trusted source. Like for example API gateway that validates
- * requested documents belongs to the right workspace.
- */
-export async function scanDocumentContent({
-  workspaceId,
+export async function getDocumentMetadata({
   document,
-  commit,
+  getDocumentByPath,
 }: {
-  workspaceId: Workspace['id']
   document: DocumentVersion
-  commit: Commit
-}): Promise<TypedResult<ConversationMetadata, LatitudeError>> {
-  const documentScope = new DocumentVersionsRepository(workspaceId)
-  const docs = await documentScope
-    .getDocumentsAtCommit(commit)
-    .then((r) => r.unwrap())
-
-  const docInCommit = docs.find((d) => d.documentUuid === document.documentUuid)
-
-  if (!docInCommit) {
-    return Result.error(
-      new UnprocessableEntityError('Document not found in commit', {}),
-    )
-  }
-
+  getDocumentByPath: (path: string) => DocumentVersion | undefined
```

Comment:
Review this change @csansoon I wanted to reuse some part of this when I already have the docs